app/src/main/java/net/squanchy/schedule/ScheduleService.kt

```diff
@@ -1,88 +1,100 @@
 package net.squanchy.schedule
 
 import io.reactivex.Observable
-import io.reactivex.functions.Function3
-import net.squanchy.schedule.domain.view.Event
+import io.reactivex.functions.BiFunction
+import io.reactivex.schedulers.Schedulers
 import net.squanchy.schedule.domain.view.Schedule
 import net.squanchy.schedule.domain.view.SchedulePage
 import net.squanchy.schedule.domain.view.Track
+import net.squanchy.schedule.filterschedule.TracksFilter
 import net.squanchy.service.firebase.FirebaseAuthService
 import net.squanchy.service.firestore.FirestoreDbService
+import net.squanchy.service.firestore.model.schedule.FirestoreEvent
 import net.squanchy.service.firestore.model.schedule.FirestoreSchedulePage
 import net.squanchy.service.firestore.toEvent
-import net.squanchy.service.firestore.toTrack
-import net.squanchy.service.repository.FilterScheduleRepository
 import net.squanchy.support.lang.Checksum
-import net.squanchy.support.lang.Optional
 import org.joda.time.DateTimeZone
 import org.joda.time.LocalDate
 
 interface ScheduleService {
-    fun schedule(onlyFavorites: Boolean): Observable<Schedule>
+    fun schedule(onlyFavorites: Boolean = false): Observable<Schedule>
     fun currentUserIsSignedIn(): Observable<Boolean>
 }
 
 class FirestoreScheduleService(
     private val authService: FirebaseAuthService,
     private val dbService: FirestoreDbService,
-    private val filterScheduleRepository: FilterScheduleRepository,
+    private val tracksFilter: TracksFilter,
     private val checksum: Checksum
 ) : ScheduleService {
 
     override fun schedule(onlyFavorites: Boolean): Observable<Schedule> {
-        return Observable.combineLatest(dbService.scheduleView(), dbService.timezone(), filterScheduleRepository.filters, combineInATriple())
```

Comment:
The flow has changed here quite a bit, now we:
 1. Get the schedule view from the `ScheduleService`
 2. Filter out events based on favourites if required (currently they're all hardcoded to `favorite = false` cause we don't have the favourites yet)
 3. Filter out events based on their track, if they have a track (so anything that doesn't have a track is never filtered out)
    a. This uses a `combineLatest` of the observable coming from the previous step and the one coming from the `TracksFilter`
 4. Once that's done we convert all the Firestore models we've been dealing with so far into domain models: `FirestoreSchedulePage` -> `SchedulePage` (and, recursively, everything inside)
    a. The schedule pages are sorted by their `date`
    b. Each schedule pages' events are sorted by their `startTime`
 5. We then `combineLatest` these with the observable of the `DateTimeZone` from `FirestoreDbService`, into pairs of `List<SchedulePage>` and `DateTimeZone`
 6. We finally unpack the pair and inject the `DateTimeZone` into the `Schedule` we create from the `SchedulePage`s

Reply:
What we don't do anymore is collecting all the tracks from the schedule (was in `extractTracksFrom()` — they now come from the DB directly)